client/me/billing-history/receipt.jsx

```diff
@@ -58,9 +62,26 @@ class BillingReceipt extends React.Component {
 			transaction,
 			transactionFetchError,
 			transactionId,
+			translate,
 		} = this.props;
 
 		if ( transactionFetchError ) {
+			if ( 'invalid_receipt' === transactionFetchError.error ) {
+				this.props.errorNotice(
+					translate( 'Could not find Receipt #%s', { args: transactionId } ),
+					{
+						displayOnNextPage: true,
+					}
+				);
+			} else {
+				this.props.clearBillingTransactionError( transactionId );
+				this.props.errorNotice( translate( 'There was a problem when fetching the receipt' ), {
```

Comment:
Looking at [other error messages](https://github.com/Automattic/wp-calypso/search?p=3&q=errorNotice(&type=&utf8=%E2%9C%93), I'd like to propose:

> Sorry, we had a problem loading that receipt.